crates/ruma-state-res/src/event_auth.rs

```diff
@@ -408,13 +413,10 @@ pub fn auth_check<E: Event>(
     if room_version.extra_redaction_checks
         && *incoming_event.event_type() == RoomEventType::RoomRedaction
     {
-        #[derive(Deserialize)]
-        struct PowerLevelsContentRedact {
-            redact: Int,
-        }
-
         let redact_level = power_levels_event
-            .and_then(|pl| from_json_str::<PowerLevelsContentRedact>(pl.content().get()).ok())
+            .and_then(|pl| {
+                deserialize_power_levels_content_redact(pl.content().get(), room_version).ok()
```

Comment:
These deserialize => `.ok()`'s are always suspicious to me. Is the right behavior really to fall back to default when part of the event fails to parse?

Reply:
This is only used on an event that should have already been validated before since it's part of the current state. So here if the deser fails it's because the field is missing, so it needs the default value.

But if you prefer I can do like in other places and log a warning if deser failed because the value has not the proper type.

Reply:
If the field is missing, `#[serde(default)]` should make it deserialize anyways. I think logging an *error* is the right solution if this should be impossible.

Reply:
It turns out, since it's a method that returns a result, we just have to return the error.